src/Sarif/Visitors/ExtractAllArtifactLocationsVisitor.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    /// <summary>
+    /// A visitor that examines a specified SARIF log to extract all artifact locations for later processing.
+    /// This will extract every location present in the log file, which may then need to be filtered by a
+    /// downstream consumer.
+    /// 
+    /// May optionally be based a run at contsruction, which allows the visitor to later visit only specific parts of
+    /// the sarif log where location data aggregation is desired.
+    /// </summary>
+    public class ExtractAllArtifactLocationsVisitor : SarifRewritingVisitor
+    {
+        private Run _currentRun;
```

Comment:
>_currentRun [](start = 20, length = 11)

Suggestion: convert this to a writable public property, CurrentRun. Remove the ctor. Set this property on VisitRun. Now users can set this and visit interior items. But the class always stays consistent. Don't forget to update the class summary if you accept this suggestion.